### PR TITLE
UN-524: Add blog + podcasts to Record Article page

### PR DESCRIPTION
**Pull request by jamesbiggs**
Ticket URL: https://national-archives.atlassian.net/browse/UN-524

## About these changes

I would've put all of the blogs + podcasts into `core` (i.e. the one in `articles` and the one in `collections`) but due to different naming conventions, this would've required extra migrations and stuff - a bit much for a temporary feature.

## How to check these changes

Go onto a `RecordArticle` page, and check that you can add up to 3 "related pages".

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: etna/collections/migrations/0008_topicexplorerpage_body.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ class Migration(migrations.Migration):
                                 (
                                     "promoted_items",
                                     wagtail.blocks.ListBlock(
-                                        etna.collections.blocks.PromotedItemBlock,
+                                        etna.collections.blocks.PromotedLinkBlock,
```
Comment (ababic) [inline etna/collections/migrations/0008_topicexplorerpage_body.py:61]:
```suggestion
                                        etna.core.blocks.PromotedLinkBlock,
```

File: etna/collections/migrations/0010_timeperiodexplorerpage_body.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ class Migration(migrations.Migration):
                                 (
                                     "promoted_items",
                                     wagtail.blocks.ListBlock(
-                                        etna.collections.blocks.PromotedItemBlock,
+                                        etna.collections.blocks.PromotedLinkBlock,
```
Comment (ababic) [inline etna/collections/migrations/0010_timeperiodexplorerpage_body.py:61]:
```suggestion
                                        etna.core.blocks.PromotedLinkBlock,
```

File: etna/collections/migrations/0018_alter_time_period_topic_index_page_body.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ class Migration(migrations.Migration):
                                 (
                                     "promoted_items",
                                     wagtail.blocks.ListBlock(
-                                        etna.collections.blocks.PromotedItemBlock,
+                                        etna.collections.blocks.PromotedLinkBlock,
```
Comment (ababic) [inline etna/collections/migrations/0018_alter_time_period_topic_index_page_body.py:69]:
```suggestion
                                        etna.core.blocks.PromotedLinkBlock,
```

File: etna/collections/migrations/0018_alter_time_period_topic_index_page_body.py [hunk 4]
```diff
@@ -131,7 +131,7 @@ class Migration(migrations.Migration):
...
```
Comment (ababic) [inline etna/collections/migrations/0018_alter_time_period_topic_index_page_body.py:134]:
```suggestion
                                        etna.core.blocks.PromotedLinkBlock,
```

File: etna/collections/migrations/0021_add_feature_insights.py [hunk 5]
```diff
@@ -68,7 +68,7 @@ class Migration(migrations.Migration):
                                 (
                                     "promoted_items",
                                     wagtail.blocks.ListBlock(
-                                        etna.collections.blocks.PromotedItemBlock,
+                                        etna.collections.blocks.PromotedLinkBlock,
```
Comment (ababic) [inline etna/collections/migrations/0021_add_feature_insights.py:71]:
```suggestion
                                        etna.core.blocks.PromotedLinkBlock,
```

File: etna/collections/migrations/0021_add_feature_insights.py [hunk 6]
```diff
@@ -113,7 +113,7 @@ class Migration(migrations.Migration):
...
```
Comment (ababic) [inline etna/collections/migrations/0021_add_feature_insights.py:116]:
```suggestion
                                        etna.core.blocks.PromotedLinkBlock,
```

File: etna/collections/migrations/0022_streamfield_use_json_field.py [hunk 7]
```diff
@@ -143,7 +143,7 @@ class Migration(migrations.Migration):
                                 (
                                     "promoted_items",
                                     wagtail.blocks.ListBlock(
-                                        etna.collections.blocks.PromotedItemBlock,
+                                        etna.collections.blocks.PromotedLinkBlock,
```
Comment (ababic) [inline etna/collections/migrations/0022_streamfield_use_json_field.py:146]:
```suggestion
                                        etna.core.blocks.PromotedLinkBlock,
```

File: etna/collections/migrations/0022_streamfield_use_json_field.py [hunk 8]
```diff
@@ -224,7 +224,7 @@ class Migration(migrations.Migration):
...
```
Comment (ababic) [inline etna/collections/migrations/0022_streamfield_use_json_field.py:227]:
```suggestion
                                        etna.core.blocks.PromotedLinkBlock,
```

File: etna/articles/models.py [hunk 9]
```diff
@@ -260,6 +261,14 @@ class RecordArticlePage(TopicalPageMixin, ContentWarningMixin, BasePageWithIntro
         help_text="Link to an external print on demand service",
     )
 
+    promoted_links = StreamField(
+        [("promoted_link", AuthorPromotedLinkBlock())],
+        max_num=3,
+        blank=True,
+        null=True,
+        use_json_field=True,
+    )
```
Comment (ababic) [inline etna/articles/models.py:270]: If this is optional, would this make sense as an option in the existing `body` `StreamField`?

Comment (jamesbiggs) [reply etna/articles/models.py:270]: Hi @ababic, there's no other StreamFields on this page - there's a "body" `MultiFieldPanel` but this is used for the stuff relevant to the `RecordArticle` such as the record info, and links to the image/print services :)

**Review by ababic — CHANGES_REQUESTED** [review]
@jamesbiggs It'd be good to include some sample output in the page template here too :)

File: templates/articles/blocks/promoted_pages.html [hunk 10]
```diff
@@ -0,0 +1,48 @@
+{% load wagtailimages_tags %}
+
+<h2>{{ value.heading }}</h2>
+<p>{{ value.sub_heading }}</p>
```
Comment (ababic) [inline templates/articles/blocks/promoted_pages.html:4]: This block appears not to exist

File: templates/articles/record_article_page.html [hunk 11]
```diff
@@ -8,6 +8,22 @@
...
+    templates/articles/blocks/promoted_pages.html will also need editing -->
+    {% if page.promoted_links %}
+        <div class="container"
+            data-container-name="topic-explorer"
+            id="analytics-topic-explorer">
+            <div class="row">
+                <div class="col-md-12">
+                    {% for block in page.promoted_links %}
+                        {% include_block block %}
+                    {% endfor %}
```
Comment (ababic) [inline templates/articles/record_article_page.html:21]: Does `{% include_block page.promoted_links %}` not work here?

**Review by ababic — APPROVED** [review]
Nice work @jamesbiggs! Approved :)

**Comment by jamesbiggs** [comment]
Gonna leave this one here until we've deployed to main on Thursday :)

**Pull request merged**